backend/src/main/java/pl/krkteam/battleships/shooting/controller/ShootingController.java

```diff
@@ -19,34 +21,42 @@
 public class ShootingController {
 
     private final ShotResultCheckerService shotResultCheckerService;
-    private final Game game;
+    private final RoomHolder roomHolder;
 
-    public ShootingController(ShotResultCheckerService shotResultCheckerService, Game game) {
+    public ShootingController(ShotResultCheckerService shotResultCheckerService,
+                              RoomHolder roomHolder) {
         this.shotResultCheckerService = shotResultCheckerService;
-        this.game = game;
+        this.roomHolder = roomHolder;
     }
 
     @PostMapping(value = "/game/player/shot")
-    public String validateShot(@RequestBody String shotJson, @RequestParam String playerName) {
+    public String validateShot(@RequestBody String shotJson, @RequestParam String playerName,
+                               @RequestParam String roomName) {
         final Player shootingPlayer = new Player(playerName);
         Gson gson = new Gson();
```

Comment:
check if you have to create a new GSON each time when you send sth to controller